spring-cloud-data-rest-resource/src/main/java/org/springframework/cloud/data/rest/resource/DetailedModuleRegistrationResource.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.rest.resource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.hateoas.PagedResources;
+
+/**
+ * Extension of {@link ModuleRegistrationResource} that contains module options
+ * and other detailed module information.
+ *
+ * @author Eric Bottard
+ * @author Gunnar Hillert
+ * @author Patrick Peralta
+ */
+public class DetailedModuleRegistrationResource extends ModuleRegistrationResource {
+
+	/**
+	 * Optional short description of the module.
+	 */
+	private String shortDescription;
+
+	/**
+	 * List of module options.
```

Comment:
If we are calling these "properties" across the board then we should rename to "options". Are we dropping the use of "options" entirely?